File: c programming/array/secondmax.c
#include<stdio.h>
int main()
{
int arr[7]={1,2,3,4,5,6,8};
   int max=arr[0];
   int smax=0;
   for(int i=0;i<7;i++)
   {
    if(arr[i]>max)
    {
      smax=max;
    max=arr[i];   
    }
    else if(arr[i]>smax && max!=arr[i])
    {
        smax=arr[i];
    }
    

   }
   printf("%d",smax); 
}

File: jAVA/array/Secondmax.java
package array;


public class Secondmax {
    public static void main(String[] args) {
        int[] arr={23,45,65,66,78,87};
        int n=arr.length;
        int mx=Integer.MIN_VALUE;
        for(int i=0;i<n;i++)
        {
           mx=Math.max(mx,arr[i]);
        }
        int smx=Integer.MIN_VALUE;
        for(int j=0;j<n;j++)
        {
            if(arr[j]!= mx)
            smx= Math.max(smx,arr[j]);
        }
        System.out.println("THe maximum value of array:"+mx);
        System.out.println("The second maximum value of array:"+smx);
    }
   
}
